numpy/lib/function_base.py

```diff
@@ -2669,6 +2689,29 @@ class vectorize(object):
            <http://docs.scipy.org/doc/numpy/reference/c-api.generalized-ufuncs.html>`_.
     """
 
+    # __new__ is overriden as a clean way to provide decorator-with-keywords
+    # syntax while allowing 'vectorize' to remain as the type of vectorized
+    # functions.
+    def __new__(cls, pyfunc=None, *args, **kwargs):
+
+        # For backwards compatibility, subclasses must not use the overloaded
+        # __new__, as vectorize.__new__ and subclass.__init__ will receive the
+        # same arguments.
+        if cls is not vectorize:
```

Comment:
I guess if we care about not breaking subclasses like those, yet want well-behaved subclasses to be able to be used as decorators, it seems there is little choice but to override `__call__`...